databases/src/connect/ConnectDatabaseAccountPromptStep.ts

```diff
@@ -0,0 +1,158 @@
+        }
+
+        return databaseItems;
+    }
+
+    public static async getCosmosDatabases(wizardContext: IConnectDBWizardContext, api: API): Promise<DBTreeItem[]> {
+
+        const cosmosClient = await createCosmosDBClient(wizardContext);
+        const cosmosAccounts: DatabaseAccountsListResult = (await cosmosClient.databaseAccounts.list())._response.parsedBody;
+        const databaseTreeItems: DBTreeItem[] = [];
+        for (const account of cosmosAccounts) {
+            const experience = tryGetExperience(account);
+            if (experience?.api === api) {
+                const accountName = nonNullProp(account, 'name');
+                const accountId = nonNullProp(account, 'id');
+                const result = (await cosmosClient.databaseAccounts.listConnectionStrings(azureUtils.azureUtils.getResourceGroupFromId(accountId), accountName))._response.parsedBody;
+
+                let connectionString: string | undefined;
+                if (experience && experience.api === "MongoDB") {
+                    const connectionStringURL: URL = new URL(nonNullProp(nonNullProp(result, 'connectionStrings')[0], 'connectionString'));
+                    // for any Mongo connectionString, append this query param because the Cosmos Mongo API v3.6 doesn't support retrywrites
+                    // but the newer node.js drivers started breaking this
+                    const searchParam: string = 'retrywrites';
+                    if (!connectionStringURL.searchParams.has(searchParam)) {
+                        connectionStringURL.searchParams.set(searchParam, 'false');
+                    }
+                    connectionString = connectionStringURL.toString();
+                    const databaseItem: DBTreeItem = { hostName: accountName, port: '5432', connectionString, azureData: { accountId: accountId, accountName: accountName, resourceGroup: azureUtils.azureUtils.getResourceGroupFromId(accountId), accountKind: getExperienceLabel(account) } };
+                    databaseTreeItems.push(databaseItem);
+
+                } else {
+                    const masterKeyResult = (await cosmosClient.databaseAccounts.listKeys(azureUtils.azureUtils.getResourceGroupFromId(accountId), accountName))._response.parsedBody;
+
+                    const endpoint = nonNullProp(account, 'documentEndpoint');
+                    const masterKey = nonNullProp(masterKeyResult, 'primaryMasterKey');
+                    const databaseItem: DBTreeItem = { hostName: accountName, port: '5432', connectionString, azureData: { accountId: accountId, accountName: accountName, resourceGroup: azureUtils.azureUtils.getResourceGroupFromId(accountId), accountKind: getExperienceLabel(account) }, docDBData: { masterKey, documentEndpoint: endpoint } };
```

Comment:
Nit: Maybe roll this one (and the other `databaseItem` above) out into multiple lines?